builtin/client/chatcommands.lua

```diff
@@ -34,3 +34,55 @@ core.register_on_sending_chat_messages(function(message)
 
 	return true
 end)
+
+core.register_chat_autocompletion(function(message, cursorpos, first_invocation)
+	if message:sub(1, 1) ~= "/" then
+		return
+	end
+	local firstspace = message:find(" ")
+	if firstspace then
+		local cmd = message:sub(2, firstspace-1)
+		local def = core.registered_chatcommands[cmd]
+		if not def then
+			core.display_chat_message("Invalid chatcommand \"" .. cmd .. "\".")
```

Comment:
Please use `-!- Invalid command` in order to be consistent with the chatcommand handling code